snaxc/transforms/convert_tosa_to_kernel.py

```diff
@@ -58,11 +66,19 @@ def match_and_rewrite(self, rescale_op: tosa.RescaleOp, rewriter: PatternRewrite
         shift = rescale_op.shift.get_values()
         assert isa(shift, tuple[int, ...])
         if isinstance(clamp_op, tosa.ClampOp):
-            max_int = assert_int8(clamp_op.max_int.value.data)
-            min_int = assert_int8(clamp_op.min_int.value.data)
+            if rescale_op.output.type.element_type == builtin.i8:
+                max_int = assert_int8(clamp_op.max_int.value.data)
+                min_int = assert_int8(clamp_op.min_int.value.data)
+            else:
+                max_int = assert_int32(clamp_op.max_int.value.data)
+                min_int = assert_int32(clamp_op.min_int.value.data)
         else:
-            max_int = 127
-            min_int = -128
+            if rescale_op.output.type.element_type == builtin.i8:
+                max_int = 127
+                min_int = -128
+            else:
+                max_int = 2147483647
+                min_int = -2147483648
```

Comment:
maybe also perhaps use xdsl instead of the magic numbers
```suggestion
                min_int, max_int = i8.value_range()
            else:
                min_int, max_int = i23.value_range()
```